lib/fork.js

```diff
@@ -59,7 +61,54 @@ class SharedWorkerChannel extends Emittery {
 
 let forkCounter = 0;
 
+const createWorker = (options, execArgv) => {
+	let worker;
+	let port;
+	let _send;
+	let _exit;
+	if (options.workerThreads) {
+		const {port1, port2} = new MessageChannel();
+		worker = new Worker(WORKER_PATH, {
+			argv: options.workerArgv,
+			env: {NODE_ENV: 'test', ...process.env, ...options.environmentVariables, AVA_PATH},
+			execArgv,
+			workerData: {
+				port: port1,
+				cwd: options.projectDir
+			},
+			transferList: [port1],
+			trackUnmanagedFds: true,
+			stdin: true,
+			stdout: true,
+			stderr: true
+		});
+		port = port2;
+		_send = (...args) => port2.postMessage(...args);
+		_exit = () => {
+			worker.terminate().then(noop, noop); // eslint-disable-line promise/prefer-await-to-then
+		};
+	} else {
+		worker = childProcess.fork(WORKER_PATH, options.workerArgv, {
+			cwd: options.projectDir,
+			silent: true,
+			env: {NODE_ENV: 'test', ...process.env, ...options.environmentVariables, AVA_PATH},
+			execArgv
+		});
+		port = worker;
+		_send = controlFlow(worker);
+		_exit = () => worker.kill();
+	}
+
+	return {
+		worker,
+		port,
+		_send,
+		_exit
```

Comment:
I used for `_exit` to avoid clashes with the `exit` used as a return value of the `fork` function, and `_send` to avoid clashes with the `send` function defined inside `fork`. It was mostly a lack of inspiration for other names 😄

Reply:
`close` and `postMessage`?

Reply:
Perfect, thanks.